src/typechecker/FStar.TypeChecker.TcEffect.fs

```diff
@@ -460,6 +460,20 @@ let tc_layered_eff_decl env0 (ed : S.eff_decl) (quals : list<qualifier>) =
   let _if_then_else_is_sound =
     let r = (ed |> U.get_layered_if_then_else_combinator |> must |> snd).pos in
 
+    (*
+     * In constructing the application nodes for subcomp and if_then_else,
+     *   we need to adjust the qualifiers
+     *
+     * Implicits remain implicits, but meta_attr or meta_arg just become implicits
+     *
+     * Don't think the boolean true below matters
+     *)
+    let binder_aq_to_arg_aq aq =
+      match aq with
+      | Some (Implicit _) -> aq
+      | Some (Meta _) -> Some (Implicit true)
```

Comment:
probably irrelevant, but may be safer to use Implicit false.

Reply:
Done.